cmd/juju/model/destroy.go

```diff
@@ -156,19 +181,78 @@ func (c *destroyCommand) Run(ctx *cmd.Context) error {
 	}
 	defer api.Close()
 
+	// Attempt to connect to the controller API.  If we can't, fail the destroy.
+	controllerAPI, err := c.getControllerAPI()
+	if err != nil {
+		return errors.Annotate(err, "cannot connect to API")
+	}
+	defer controllerAPI.Close()
+
 	// Attempt to destroy the model.
+	ctx.Infof("Destroying model")
 	err = api.DestroyModel(names.NewModelTag(modelDetails.ModelUUID))
 	if err != nil {
 		return c.handleError(errors.Annotate(err, "cannot destroy model"), modelName)
 	}
 
+	// Wait for model to be destroyed.
+	const modelStatusPollWait = 2 * time.Second
+	modelStatus := newTimedModelStatus(ctx, controllerAPI, names.NewModelTag(modelDetails.ModelUUID), c.sleepFunc)
+	modelData := modelStatus(0)
+	for modelData != nil {
+		ctx.Infof(formatDestroyModelInfo(modelData) + "...")
+		modelData = modelStatus(modelStatusPollWait)
+	}
+
 	err = store.RemoveModel(controllerName, modelName)
 	if err != nil && !errors.IsNotFound(err) {
 		return errors.Trace(err)
 	}
 	return nil
 }
 
+type modelData struct {
+	machineCount     int
+	applicationCount int
+}
+
+// newTimedModelStatus returns a function which waits a given period of time
+// before querying the API server for the status of a model.
+func newTimedModelStatus(ctx *cmd.Context, api ModelStatusAPI, tag names.ModelTag, sleepFunc func(time.Duration)) func(time.Duration) *modelData {
+	return func(wait time.Duration) *modelData {
+		sleepFunc(wait)
+		status, err := api.ModelStatus(tag)
+		if err != nil {
+			if params.ErrCode(err) != params.CodeNotFound {
```

Comment:
I was thinking more that ErrPerm might be returned once the model is removed. Presumably you tested it, and it would have shown up if it was a problem.